backend/src/schema/resolvers/reports.spec.js

```diff
@@ -1,521 +1,599 @@
-            createdAt: expect.any(String),
+      await Promise.all([
+        factory.create('Post', {
+          author: abusiveUser,
+          id: 'abusive-post-1',
+          categoryIds,
+          content: 'Interesting Knowledge',
+        }),
+        factory.create('Post', {
+          author: moderator,
+          id: 'post-2',
+          categoryIds,
+          content: 'More things to do …',
+        }),
+        factory.create('Post', {
+          author: currentUser,
+          id: 'post-3',
+          categoryIds,
+          content: 'I am at school …',
+        }),
+      ])
+      await Promise.all([
+        factory.create('Comment', {
+          author: currentUser,
+          id: 'abusive-comment-1',
+          postId: 'post-1',
+        }),
+      ])
+      authenticatedUser = await currentUser.toJson()
+      await Promise.all([
+        mutate({
+          mutation: reportMutation,
+          variables: {
+            resourceId: 'abusive-post-1',
             reasonCategory: 'other',
             reasonDescription: 'This comment is bigoted',
-            submitter: expect.objectContaining({
-              id: 'user1',
-            }),
-            type: 'Post',
-            user: null,
-            post: expect.objectContaining({
-              id: 'p1',
-            }),
-            comment: null,
-          }),
-          expect.objectContaining({
-            createdAt: expect.any(String),
+          },
+        }),
+        mutate({
+          mutation: reportMutation,
+          variables: {
+            resourceId: 'abusive-comment-1',
             reasonCategory: 'discrimination_etc',
             reasonDescription: 'This post is bigoted',
-            submitter: expect.objectContaining({
-              id: 'user1',
+          },
+        }),
+        mutate({
+          mutation: reportMutation,
+          variables: {
+            resourceId: 'abusive-user-1',
+            reasonCategory: 'doxing',
+            reasonDescription: 'This user is harassing me with bigoted remarks',
+          },
+        }),
+      ])
+      authenticatedUser = null
+    })
+    describe('unauthenticated', () => {
+      it('throws authorization error', async () => {
+        authenticatedUser = null
+        expect(query({ query: reportsQuery })).resolves.toMatchObject({
+          data: { reports: null },
+          errors: [{ message: 'Not Authorised!' }],
+        })
+      })
+    })
+    describe('authenticated', () => {
+      it('role "user" gets no reports', async () => {
+        authenticatedUser = await currentUser.toJson()
+        expect(query({ query: reportsQuery })).resolves.toMatchObject({
+          data: { reports: null },
+          errors: [{ message: 'Not Authorised!' }],
+        })
+      })
+
+      it('role "moderator" gets reports', async () => {
+        const expected = {
+          // to check 'orderBy: createdAt_desc' is not possible here, because 'createdAt' does not differ
```

Comment:
it is possible, the `createdAt` just needs to be updated... I know you didn't add this test/comment @aonomike, but if you did want to test the order, there is an example of how to do that in my `PR` here:
https://github.com/Human-Connection/Human-Connection/pull/1840/files#diff-4da0f183f4d13eb4a864f960e8267785R771

Reply:
created a ticket for this